File: 071.LeetCode@101_Symmetric_Tree/071.LeetCode@101_Symmetric_Tree.java
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root ==null) return true;
        return helper(root.left,root.right);
    }
    public boolean helper (TreeNode root1,TreeNode root2){
        if (root1==null && root2==null ) return true;
        else if (root1==null || root2==null ) return false;
        if (root1.val!=root2.val){return false;}
        return helper(root1.left,root2.right)&&helper(root1.right,root2.left);
    }
}

File: 071.LeetCode@101_Symmetric_Tree/071.LeetCode@101_Symmetric_Tree.py
class Solution:
    def isSymmetric(self, root: [TreeNode]):
        if not root:
            return True
        return self.helper(root.left, root.right)

    def helper(self, root1, root2):
        if not root1 and not root2:
            return True
        if not root1 or not root2:
            return False
        if root1.val != root2.val:
            return False
        return self.helper(root1.left, root2.right) and self.helper(root1.right, root2.left)